lib/chef/resource/aws_subnet.rb

```diff
@@ -75,6 +76,12 @@ class Chef::Resource::AwsSubnet < Chef::Provisioning::AWSDriver::AWSResourceWith
   #
   attribute :route_table#, kind_of: [ String, AwsRouteTable, AWS::EC2::RouteTable ], equal_to: [ :default_to_main ]
 
+  #
+  # The Network ACL to associate with this subnet. Subnets may only
+  # be associated with a single Network ACL.
+  #
+  attribute :network_acl, kind_of: [ String, AwsNetworkAcl, AWS::EC2::NetworkACL ]
```

Comment:
Another issue that has come up recently is the ability to remove references.  IE, how do I unlink a network_acl from a subnet?  I'm going to propose that we introduce the idea of adding the attribute value `network_acl false` as the signifier for unlinking the two.  How does this sound?

If we add this we should also add a test covering this.

Paging @jkeiser @randomcamel @metadave